docs/pipeline.rst

```diff
@@ -44,6 +44,59 @@ The resulting graph is:
 
 .. image:: images/two_readers.svg
 
+.. _processing_graph_structure:
+
+Processing Graph Structure
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+DALI pipelines are executed in stages. The stages correspond to the ``device`` parameter
+that can be specified for the operator, and are executed in following order:
+
+#. ``'cpu'`` - operators that accept CPU inputs and produce CPU outputs.
+#. ``'mixed'`` - operators that accept CPU inputs and produce GPU outputs, for exampe :meth:`nvidia.dali.fn.decoders.image`.
+#. ``'gpu'`` - operators that accept GPU inputs and produce GPU outpus.
+
+Additionaly, ``.gpu()`` might be used on a :class:`~nvidia.dali.pipeline.DataNode` (the output of
+the operator in pipeline definition) to transfer data from CPU to GPU.
+
+Once tha data is transferred to GPU it can't be transferred back to CPU within the DALI pipeline.
```

Comment:
```suggestion
Data that has been produced by a later stage cannot be consumed by an operator executing in an earlier stage.
```
The original text suggests that transfer is somehow destructive, like in:
```Python
a = some_cpu_op()
some_gpu_op(a.gpu())
# `a` is unusable on CPU any more
```

Reply:
Done